### PR TITLE
update EnvironmentCredential and azure.core.util.Configuration

### PR DESCRIPTION
**Pull request by cxznmhdcxz**
Updated EnvironmentCredential to support env vars for UsernamePasswordCredential and ClientCertificateCredential. Added PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH to azure.core.util.Configuration.

same code as https://github.com/Azure/azure-sdk-for-java/pull/7337, as https://github.com/Azure/azure-sdk-for-java/pull/7337 was reverted

### CODE REVIEW

**Comment by alzimmermsft** [comment]
This build is failing since Identity is relying on the released version of Azure Core and these changes depend on source code Azure Core, this is why the Identity from source validation job is passing but all others are failing. To fix this the dependency version of Azure Core will need to be changed to `unreleased_com.azure:azure-core`.

**Comment by g2vinay** [comment]
Rebase with master.
and update core dependency to this

> 
>     <dependency>
>       <groupId>com.azure</groupId>
>       <artifactId>azure-core</artifactId>
>       <version>1.4.0-beta.1</version> 
>     </dependency>

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java [hunk 1]
```diff
@@ -14,13 +14,27 @@
...
  * </ul>
+ * or:
+ * <p>
+ * <ul>
+ *     <li>{@link Configuration#PROPERTY_AZURE_CLIENT_ID AZURE_CLIENT_ID}</li>
+ *     <li>{@link Configuration#PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH AZURE_CLIENT_CERTIFICATE_PATH}</li>
+ *     <li>{@link Configuration#PROPERTY_AZURE_TENANT_ID AZURE_TENANT_ID}</li>
+ * </ul>
+ * or:
+ * <p>
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java:32]: the <p> tag is not closed.
Is it needed here ?
you don't have it around other <ul> elements.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java [hunk 2]
```diff
@@ -40,15 +54,22 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
+        String clientId = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_ID);
+        String tenantId = configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID);
+        String clientSecret = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_SECRET);
+        String certPath = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_CERTIFICATE_PATH);
+        String username = configuration.get(Configuration.PROPERTY_AZURE_USERNAME);
+        String password = configuration.get(Configuration.PROPERTY_AZURE_PASSWORD);
```
Comment (srnagar) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/EnvironmentCredential.java:62]: This should be within the supplier to get the configuration asynchronously and when getToken() is subscribed. This fetches the configuration synchronously and immediately.

**Pull request merged**